code-challenges.test.js

```diff
@@ -21,9 +21,38 @@ const people = [
 ]
 // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
 
+describe("sentence", () => {
+  it("takes in an array of objects and returns an array with a sentence about each person with their name capitalized.", () => {
+    expect(sentence(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+  })
+})
 
 // b) Create the function that makes the test pass.
 
+// Pseudo Code:
+// Create a function that takes an array as a parameter. (sentence)
+// Create a variable of an empty array. (newArr)
+// Create a variable for the first key value of the 0 index of the array. (name1)
+// Create a variable for the first key value of the 1 index of the array. (name2)
+// Create a variable for the first key value of the 2 index of the array. (name3)
+// Use a method, with string interpolation, to add the variables to the empty array variable (.push)
+  // Within push, use built in methods to capitalize the first letter of each name and add them to a substring of the remainder of the name. (charAt, toUpperCase, substring)
+  // Within push, use string interpolation to represent the occupation key of the given index of the array. 
+// Have the function return the new array variable
+// Run the function with the provided variables
+
+function sentence(arr) {
+  let newArr = [];
+  let name1 = Object.values(arr[0])[0];
+  let name2 = Object.values(arr[1])[0];
+  let name3 = Object.values(arr[2])[0];
+  newArr.push(`${name1.charAt(0).toUpperCase() + name1.substring(1, 5) + name1.charAt(5).toUpperCase() + name1.substring(6)} is ${Object.values(arr[0])[1]}`)
+  newArr.push(`${name2.charAt(0).toUpperCase() + name2.substring(1, 7) + name2.charAt(7).toUpperCase() + name2.substring(8)} is ${Object.values(arr[0])[1]}`)
+  newArr.push(`${name3.charAt(0).toUpperCase() + name3.substring(1, 7) + name3.charAt(7).toUpperCase() + name3.substring(8)} is ${Object.values(arr[0])[1]}`)
+  return newArr
+}
+
+console.log(sentence(people))
```

Comment:
Hi Sam - this was a difficult one!  Your pseudocode is beautiful! I notice a few things in your code.  It does pass the test, however it isn't dynamic in the sense that it won't pass if we change the input. Also, you are declaring the function with the `function` keyword instead of the ES6 syntax. Last thing is that you are combining String Interpolation and concatenation.  I've never seen you do any of these before, so I suggest you revert back to your previous syntax and process.  You are a great developer - I encourage you to stretch yourself and try to use a HOF to refactor this code. I know you can do it!